File: 03_greedy/Hoon/leetcode_1221.js
const balancedStringSplit = (s) => {
  let array = s.split('');
  let currentTerm = array.shift();
  let countSameWord = 1;
  let isNewTerm = false;
  let result = 0;

  while (array.length) {
    if (currentTerm === array[0] || isNewTerm) {
      isNewTerm = false;
      countSameWord++;
      array.shift();
    } else {
      countSameWord--;

      if (countSameWord === 0) {
        isNewTerm = true;
        currentTerm = array[1];
        result++;
      }

      array.shift();
    }
  }

  return result;
};

File: 03_greedy/Aiden/leetcode_1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        left = right = result = 0
        
        for c in s:
            if c == 'L':
                left +=1
            else:
                right +=1
                
            if left == right:
                left = right = 0
                result +=1
        
        return result